src/components/scenes/EditTokenScene.tsx

```diff
@@ -187,6 +190,27 @@ function EditTokenSceneComponent(props: Props) {
     }
   })
 
+  const autoCompleteToken = async (searchString: string) => {
+    // Ignore autocomplete if the scene was initialized with any of the token details prefilled,
+    if (route.params.currencyCode != null || route.params.displayName != null || route.params.multiplier != null || route.params.networkLocation != null) {
+      return
+    }
+
+    const [token] = await wallet.currencyConfig.getTokenDetails({ contractAddress: searchString })
```

Comment:
We need some way to rate-limit this call, so we don't spam the network as the user types. The simplest way is a busy flag, so if one call is pending, we don't trigger any more until after that one finishes.

Reply:
I added a pretty basic rate limiter. There are obvious race conditions but since this is a field the user would most likely paste a value in I'm not concerned about it.